utils/nctl/sh/scenarios/sync_upgrade_test.sh

```diff
@@ -0,0 +1,220 @@
+    log "------------------------------------------------------------"
+
+    do_await_genesis_era_to_complete
+
+    # 1. Send batch of Wasm deploys
+    do_send_wasm_deploys
+    # 2. Send batch of native transfers
+    do_send_transfers
+    # 3. Wait until they're all included in the chain.
+    do_await_deploy_inclusion
+    # 4. Upgrade the network
+    do_upgrade_network
+    # 5. Wait for the network to upgrade.
+    do_await_network_upgrade
+    # 6. Take a note of the last finalized block hash
+    do_read_lfb_hash
+    # 7. Send batch of Wasm deploys
+    do_send_wasm_deploys
+    # 8. Send batch of native transfers
+    do_send_transfers
+    # 9. Wait until they're all included in the chain.
+    do_await_deploy_inclusion
+    # 10. Start the node in sync mode using hash from 4)
+    do_start_new_node "$NEW_NODE_ID"
+    # 11. Wait until it's synchronized
+    #     and verify that its last finalized block matches other nodes'.
+    do_await_full_synchronization "$NEW_NODE_ID"
+
+    log "------------------------------------------------------------"
+    log "Syncing node complete"
+    log "------------------------------------------------------------"
+}
+
+function log_step() {
+    local COMMENT=${1}
+    log "------------------------------------------------------------"
+    log "STEP $STEP: $COMMENT"
+    log "------------------------------------------------------------"
+    STEP=$((STEP + 1))
+}
+
+function do_await_genesis_era_to_complete() {
+    log_step "awaiting genesis era to complete"
+    while [ "$(get_chain_era)" != "1" ]; do
+        sleep 1.0
+    done
+}
+
+function do_upgrade_network() {
+    log_step "scheduling the network upgrade to version ${PROTOCOL_VERSION} at era ${ACTIVATE_ERA}"
+    for NODE_ID in $(seq 1 "$(get_count_of_nodes)"); do
+        _upgrade_node "$PROTOCOL_VERSION" "$ACTIVATE_ERA" "$NODE_ID"
+    done
+}
+
+function do_await_network_upgrade() {
+    log_step "wait for the network to upgrade"
+    local WAIT_UNTIL=$((ACTIVATE_ERA + 2))
+    while [ "$(get_chain_era)" != "$WAIT_UNTIL" ]; do
+        sleep 1.0
+    done
+}
+
+function do_send_wasm_deploys() {
+    # NOTE: Maybe make these arguments to the test?
+    local BATCH_COUNT=1
+    local BATCH_SIZE=1
+    local TRANSFER_AMOUNT=10000
+    log_step "sending Wasm deploys"
+    # prepare wasm batch
+    prepare_wasm_batch "$TRANSFER_AMOUNT" "$BATCH_COUNT" "$BATCH_SIZE"
+    # dispatch wasm batches
+    for BATCH_ID in $(seq 1 $BATCH_COUNT); do
+        dispatch_wasm_batch "$BATCH_ID"
+    done
+}
+
+function do_send_transfers() {
+    log_step "sending native transfers"
+    # NOTE: Maybe make these arguments to the test?
+    local AMOUNT=2500000000
+    local TRANSFERS_COUNT=5
+    local NODE_ID="random"
+
+    # Enumerate set of users.
+    for USER_ID in $(seq 1 "$(get_count_of_users)"); do
+        dispatch_native "$AMOUNT" "$USER_ID" "$TRANSFERS_COUNT" "$NODE_ID"
+    done
+}
+
+function do_await_deploy_inclusion() {
```

Comment:
function name is a bit misleading since its not actually checking the deploys were included

Reply:
I know but tracking the inclusion status of randomly generated deploys is going to be a pain. It's supposed to be tested with other WG-based tests. Here I just want to "pretend" they are included, which they should with nctl parameters. It's also not critical that they're all included. It's enough if some of them are. I could call `await_n_eras 1` but that wouldn't convey why I'm doing it. Hence this method.